### PR TITLE
Fix render order of overlays

### LINKED ISSUE #6001: Soft overlay does not always render in-world
If I define several nodes that use different base images, but all of which use the same single overlay image, the engine will only render the overlay texture on one of them.  The rest all show only their base images, as if no overlay is defined.

This only affects the nodes as displayed in-world -- the overlays ARE rendered properly in the hotbar, inventory, and in the Craft Guide/Creative grid on the right in Unified Inventory.

Suppose I have four nodes.  Each one has its own texture (let's just say they're cobble, wood, stone, and sand).  All four of them reference the same overlay texture file, let's call it "foo_overlay.png".  Whichever one is registered last seems to be the one that will show the overlay.

If I make a copy of the overlay, e.g. `cp foo_overlay.png foo_overlay_2.png`, and I reference the copy in one of the nodes that isn't displaying its overlay, then it works.  If I make additional copies of the texture file and reference those copies in the other incorrectly-displayed nodes, then those nodes start working as well.

To see this in practice, download [Unified Bricks](https://github.com/minetest-mods/unifiedbricks).  Build a wall about 20x10 in size, using all four brick patterns (plain, multicolor light, multicolor medium, and multicolor dark).

Now open `init,lua` in a text editor, and look for the mortar overlays referenced on lines 101, 156, 177, and 198.  Change them all to the same mortar overlay file, then try again.  The wall will lose its overlay textures on some of its nodes.

screenshot_2017-06-16_17-08-53

On the left, all four node types have a unique filename for the overlay (though they are all exact copies).  On the right is what happens if all four nodes are pointed at the same overlay file (e.g. one copy).

### PR DESCRIPTION
**Pull request by juhdanad**
Fixes #6001.

The overlays were rendered in the wrong order if they were used  with multiple nodes.

### CODE REVIEW

File: src/clientmap.cpp [hunk 1]
```diff
@@ -283,49 +283,48 @@ void ClientMap::updateDrawList(video::IVideoDriver* driver)
...
 	}
 
 	void add(scene::IMeshBuffer *buf, u8 layer)
 	{
+		// Append to the correct layer
+		std::vector<MeshBufList> &list = lists[layer];
 		const video::SMaterial &m = buf->getMaterial();
-		for(std::vector<MeshBufList>::iterator i = lists.begin();
-				i != lists.end(); ++i){
+		for(std::vector<MeshBufList>::iterator i = list.begin();
```
Comment (nerzhul) [inline src/clientmap.cpp]: best, replace this loop with a C++11 loop for (MesBufList &l : list) { }

File: src/clientmap.cpp [hunk 2]
```diff
@@ -469,35 +468,38 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
...
-				infostream << "ClientMap::renderMap(): "
-					"Rendering takes ages, returning."
-					<< std::endl;
-				return;
+	// Render all layers in order
+	for (int layer = 0; layer < MAX_TILE_LAYERS; layer++) {
+		std::vector<MeshBufList> &lists = drawbufs.lists[layer];
+
+		int timecheck_counter = 0;
+		for (std::vector<MeshBufList>::iterator i = lists.begin();
```
Comment (nerzhul) [inline src/clientmap.cpp]: same here

File: src/clientmap.cpp [hunk 3]
```diff
@@ -469,35 +468,38 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
...
+			driver->setMaterial(list.m);
 
-		for (std::vector<scene::IMeshBuffer*>::iterator j = list.bufs.begin();
-				j != list.bufs.end(); ++j) {
-			scene::IMeshBuffer *buf = *j;
-			driver->drawMeshBuffer(buf);
-			vertex_count += buf->getVertexCount();
-			meshbuffer_count++;
-		}
+			for (std::vector<scene::IMeshBuffer*>::iterator j = list.bufs.begin();
```
Comment (nerzhul) [inline src/clientmap.cpp]: same here

**Comment by juhdanad** [comment]
The C++11 formatting is done.

File: src/clientmap.cpp [hunk 4]
```diff
@@ -469,35 +465,32 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
...
+	// Render all layers in order
+	for (int layer = 0; layer < MAX_TILE_LAYERS; layer++) {
+		std::vector<MeshBufList> &lists = drawbufs.lists[layer];
+
+		int timecheck_counter = 0;
+		for (MeshBufList &list : lists) {
+			timecheck_counter++;
+			if (timecheck_counter > 50) {
+				timecheck_counter = 0;
+				int time2 = time(0);
```
Comment (Zeno-) [inline src/clientmap.cpp]: Is int the appropriate type?

Comment (nerzhul) [inline src/clientmap.cpp]: please use std::time_t

**Comment by Zeno-** [comment]
:+1:

**Pull request merged**